### PR TITLE
Assignment4 

### PR DESCRIPTION
**Pull request by roonle**
Missing Group and Gift Controller test and more unit testing

### CODE REVIEW

**Comment by JerettLatimer** [comment]
Will Review!

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 1]
```diff
@@ -2,37 +2,61 @@
...
+            .UseSqlite(sqliteConnection));
 
-        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public static void Configure(IApplicationBuilder app, IWebHostEnvironment env)
-        {
-            if (env.IsDevelopment())
-            {
-                app.UseDeveloperExceptionPage();
-            }
+            services.AddScoped<IGiftService, GiftService>();
```
Comment (JerettLatimer) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Correctly registered services.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 2]
```diff
@@ -2,37 +2,61 @@
...
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseRouting();
+
+            app.UseOpenApi();
+            
+            app.UseSwaggerUi3();
```
Comment (JerettLatimer) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Correct NSwag implementation.

File: SecretSanta/src/SecretSanta.Api/Startup.cs [hunk 3]
```diff
@@ -2,37 +2,61 @@
...
     #pragma warning restore CA1052 // Static holder types should be Static or NotInheritable
     {
-        // This method gets called by the runtime. Use this method to add services to the container.
-        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
-        //public void ConfigureServices(IServiceCollection services)
-        //{
-        //}
+        public static void ConfigureServices(IServiceCollection services)
+        {
+#pragma warning disable CA2000 // connection is used further out
```
Comment (JerettLatimer) [inline SecretSanta/src/SecretSanta.Api/Startup.cs]: Good job disabling this warning correctly.

**Comment by JerettLatimer** [comment]
Your assignment looks good so far. You just need to complete your unit tests!

File: SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            var controller = new GiftController(service);
+
+            // Act
+            ActionResult<Gift> rv = await controller.Get(Gift.Id);
+
+            // Assert
+            Assert.IsTrue(rv.Result is OkObjectResult);
+        }
+
+    }
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/GiftControllerTests.cs]: Would have like to see the other controller methods tested.

**Pull request closed without merging**